tests/integration_tests/test_hyperopt.py

```diff
@@ -313,5 +383,57 @@ def test_hyperopt_run_hyperopt(csv_filename, search_space, tmpdir, ray_cluster):
     # check for existence of the hyperopt statistics file
     assert os.path.isfile(os.path.join(tmpdir, "test_hyperopt", "hyperopt_statistics.json"))
 
-    if os.path.isfile(os.path.join(tmpdir, "test_hyperopt", "hyperopt_statistics.json")):
-        os.remove(os.path.join(tmpdir, "test_hyperopt", "hyperopt_statistics.json"))
+
+@pytest.mark.distributed
+def test_hyperopt_run_shared_params_trial_table(csv_filename, tmpdir):
+    config, rel_path, num_filters_search_space, embedding_size_search_space = _setup_ludwig_config_with_shared_params(
+        csv_filename
+    )
+
+    hyperopt_results = hyperopt(config, dataset=rel_path, output_directory=tmpdir, experiment_name="test_hyperopt")
+    hyperopt_results_df = hyperopt_results.experiment_analysis.results_df
+
+    # Check that the trials did sample from defaults in the search space
+    for _, trial_row in hyperopt_results_df.iterrows():
+        embedding_size = _get_trial_parameter_value("defaults.output_features.category.embedding_size", trial_row)
+        num_filters = _get_trial_parameter_value("defaults.input_features.text.num_filters", trial_row)
+        assert embedding_size in embedding_size_search_space
+        assert num_filters in num_filters_search_space
+
+
+@pytest.mark.distributed
+def test_hyperopt_with_shared_params_written_config(csv_filename, tmpdir):
+    config, rel_path, num_filters_search_space, embedding_size_search_space = _setup_ludwig_config_with_shared_params(
```

Comment:
Can you consolidate this test and `test_hyperopt_run_shared_params_trial_table` into a single test?

It seems like they run the same hyperopt experiment (which is kind of slow to run). If you run the experiment only once and test everything on the same experiment, the tests will be faster overall.

I would recommend keeping the abstraction you've used here, but merging the tests. So something like:

```
def test_shared_parameters_trial_table(hyperopt_results_df, shared_params_dict):
    # Checks that the trials did sample from defaults in the search space
    # Shared params dict: {shared_param_name: shared_param_search_space}. E.g.: {'defaults.output_features.category.embedding_size': embedding_size_search_space, ...}
    for _, trial_row in hyperopt_results_df.iterrows():
        for shared_param_name, shared_param_search_space in shared_params_dict.items():
            param_val = _get_trial_parameter_value(shared_param_name, trial_row)
            assert param_val in shared_param_search_space


def test_shared_params_written_config(hyperopt_results_df, ...):
    # Abstracts out everything that happens from lines 414 - 439


@pytest.mark.distributed
def test_hyperopt_with_shared_params(csv_filename, tmpdir):
    config, rel_path, num_filters_search_space, embedding_size_search_space = _setup_ludwig_config_with_shared_params(
        csv_filename
    )

    hyperopt_results = hyperopt(config, dataset=rel_path, output_directory=tmpdir, experiment_name="test_hyperopt")
    hyperopt_results_df = hyperopt_results.experiment_analysis.results_df

    test_shared_parameters_trial_table(...)
    test_shared_params_written_config(...)

```

Reply:
Makes sense to me, will update!